### ISSUE TITLE
'VALIDATE_ALL_CODEBASE: false' does not work

### ISSUE THREAD

**Issue by dshevtsov** [github/super-linter#1123]
## Describe the bug

In my workflow, I want to scan only new or edited files, hence I have had set `VALIDATE_ALL_CODEBASE: false` ([the setting](https://github.com/magento/devdocs/blob/2.4.2-develop/.github/workflows/linter.yml#L47)).

When the workflow runs on a pull request (PR) that contains changes to `.md` files only, it fails on `.js` files that have not even been expected to be scanned, because the PR does not contain any changes to `.js` files.

## Expected behavior

When the workflow with the `VALIDATE_ALL_CODEBASE: false` settings runs on a pull request, it scans only new or edited files as stated in technical documentation.

## Steps to Reproduce

Adding a link to real case at public repository in the next clause.

## Logs

https://github.com/magento/devdocs/pull/8474/checks?check_run_id=1651122030

**Comment by GaboFDC** [comment]
Maybe this is a case of documentation not been clear or missing.
The way we know what files to check when `VALIDATE_ALL_CODEBASE=false` is to compare the current branch with the `DEFAULT_BRANCH`. So in this case the target of the pr and the default branch are different, so the changed files, are not the same.

**Comment by dshevtsov** [comment]
> Maybe this is a case of documentation not been clear or missing.

@GaboFDC yes, I think this is the case. My expectations were that the tool would've checked changes between the Base branch and the Head branch of the pull request.

So basically the tool allows testing only pull requests to a single branch which is the default branch of the repository.
I cannot use the tool for changes between other branches. Correct?

**Comment by iainlane** [comment]
Is this a restriction that is part of the design of super linter, or would you accept a PR to change this to check out the target branch? I'm new to actions, but from reading the docs it looks like `github.base_ref` could be used as a source to diff against, for pull requests.

**Comment by BretFisher** [comment]
OK I'm back with a full (yet slightly complex) solution for this DEFAULT_BRANCH limitation until we get a built-in solution for Super-Linter.

This uses steps to dynamically decide if it needs to set DEFAULT_BRANCH to a PR base, or to the existing branch. That way, you can PR to any branch and this will do the correct file diff:

TL;DR version without comments and extra troubleshooting echo's:

```yaml
    steps:
      - name: Checkout Code
        uses: actions/checkout@v2.3.4
        with:
          fetch-depth: 0

      - name: Set DEFAULT_BRANCH to PR target
        if: ${{ github.base_ref != '' }}
        run: echo "DEFAULT_BRANCH=${{ github.base_ref }}" >> $GITHUB_ENV

      - name: Set DEFAULT_BRANCH to current branch
        if: ${{ github.base_ref == '' }}
        run: echo "DEFAULT_BRANCH=$(echo '${{ github.ref }}' | sed 's/refs\/heads\///')" >> $GITHUB_ENV

      - name: Lint Code Base
        uses: github/super-linter@v4.2.2
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          DEFAULT_BRANCH: ${{ env.DEFAULT_BRANCH }}
          VALIDATE_ALL_CODEBASE: fals
```

Same version, but with a bunch of comments and a debug step to dump out various values:

```yaml
    steps:
      # Checkout the code base
      - name: Checkout Code
        uses: actions/checkout@v2.3.4
        with:
          # Full git history is needed to get a proper list of changed files within `super-linter`
          fetch-depth: 0

      - name: Set DEFAULT_BRANCH to PR target
        # if base_ref has a value, this is a PR
        # we save the PR target branch name to a variable for use in linter config
        # we pass string between job steps by echoing to $GITHUB_ENV, making it available in $env later
        if: ${{ github.base_ref != '' }}
        run: echo "DEFAULT_BRANCH=${{ github.base_ref }}" >> $GITHUB_ENV

      - name: Set DEFAULT_BRANCH to current branch
        # if base_ref has no value, this is just a commit on a branch
        # we need to strip refs/heads from github.ref to find the current branch name
        # then save the current branch name to a variable for use in linter config later
        # we pass strings between job steps by echoing to $GITHUB_ENV, making it available in $env later
        if: ${{ github.base_ref == '' }}
        run: echo "DEFAULT_BRANCH=$(echo '${{ github.ref }}' | sed 's/refs\/heads\///')" >> $GITHUB_ENV

      - name: Echo out branch values
        run: | 
          echo "${{ env.DEFAULT_BRANCH }}"
          echo github.base_ref: ${{ github.base_ref }}
          echo github.head_ref: ${{ github.head_ref }}
          echo github.ref: ${{ github.ref }}
      # Example output: in a PR branch (head) named `feature-1 to a target branch (base) of `release-v1`, you'll see:
        # github.base_ref: release-v1
        # github.head_ref: feature-1
        # github.ref: refs/pull/2/merge
      # if github.base_ref is empty, then commit is against a branch with no PR
        # this could have been a commit on any branch
        # In that case, github.ref show the branch we are on: refs/heads/main
        # DEFAULT_BRANCH: regex to match branch name (remove "refs/heads/" prefix)
      # elseif github.base_ref is not empty, then commit is against a branch with a PR target
        # github.base_ref will have the target branch name
        # github.head_ref will have the branch name we are on
        # github.ref will have the PR number: refs/pull/1/merge
        # DEFAULT_BRANCH: ${{ github.base_ref }}

      # Run Linter against code base
      # https://github.com/marketplace/actions/super-linter
      - name: Lint Code Base
        uses: github/super-linter@v4.2.2
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          # the value needs to change when Git default branch changes
          DEFAULT_BRANCH: ${{ env.DEFAULT_BRANCH }}
          # setting this to false means that only changed files will be scanned in each commit
          VALIDATE_ALL_CODEBASE: false
```
